### ISSUE TITLE
Refactor linalg::svd to also compute V

### ISSUE THREAD

**Issue by shubham808** [shogun-toolbox/shogun#4368]
Any ```m x n``` matrix A can be factored into ```A = USVt```. Currently [linalg::svd](https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/mathematics/linalg/LinalgNamespace.h#L1635) only computes U and s. There should also be some API for computing V matrix to cover all use cases(see [here](https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/mathematics/linalg/backend/eigen/BasicOps.cpp#L258)).

**Comment by gf712** [comment]
This should be done with the new linalg API

**Comment by Khalifa1997** [comment]
@gf712 I'd like to work on this, any suggestions where I should start^^

**Comment by gf712** [comment]
@Khalifa1997 this is not a priority atm as we are looking into porting all of this to a new API which is under heavy development. I also don't know if there is any use case currently.

**Comment by Khalifa1997** [comment]
@gf712  Interesting, would you recommend any issues that maybe related to using the new linalg, I haven't worked on it and maybe this could be a good start to me to dive deeper into it

**Comment by gf712** [comment]
The new linalg module is still being built, so there is not much that can be done there.
I am not sure about specific issues, but you can have a look through the library and find several places where we use raw pointers and do operations on that. But ideally it should all be done using SGVector/SGMatrix together with the operations provided by the current linalg module. That will be helpful, because it will be easier to port to the new module, once it is done :)

**Comment by Khalifa1997** [comment]
@gf712 Great! thanks I will look into it more tomorrow
